File: public/javaCode/selectionsort.js
function selectionSort(numbers) {
    const n = numbers.length;

    for (let i = 0; i < n; i++) {
        let minIndex = i;

        for (let j = i + 1; j < n; j++) {
            if (numbers[j] < numbers[minIndex]) {
                minIndex = j;
            }
        }


        [numbers[i], numbers[minIndex]] = [numbers[minIndex], numbers[i]];
    }
}

// Number of elements
const amount = 10000;

const arr = Array.from({ length: amount }, () =>
    Math.floor(Math.random() * 1000) + 1
);

const start = performance.now();

selectionSort(arr);

const end = performance.now();

console.log(arr);
console.log(`Time taken: ${(end - start).toFixed(6)} ms`);

File: public/pythonCode/selectionsort.py
import time
import random

def selectionSort(numbers: list[int]) -> None:
    n: int = len(numbers)
    for i in range(n):
        minIndex: int = i
        for j in range(i + 1, n):
            if numbers[j] < numbers[minIndex]:
                minIndex = j
        numbers[i], numbers[minIndex] = numbers[minIndex], numbers[i]

amount = 10000
# You can change this to see how long it take to completely sort the array
arr = [random.randint(1, 1000) for i in range(amount)]
start = time.time()
selectionSort(arr)
end = time.time()

print(arr)
print(f"Time taken: {end - start:.6f} seconds")